build-tools/lib/pulumi-helpers

```diff
@@ -123,6 +127,24 @@ function _pulumi_automation_run_up() {
     fi
 }
 
+function _remove_pulumi_finalizers() {
+    for resource in stack update; do
+      kubectl get "$resource" --all-namespaces -o json | jq -r '.items[] | "\(.metadata.namespace) \(.metadata.name)"' | while read -r namespace name; do
+        finalizers=$(kubectl get "$resource" -n "$namespace" "$name" -o json | jq -r '.metadata.finalizers')
+        if [[ "$finalizers" == "null" ]]; then
+          echo "$resource $name in namespace $namespace has no finalizers, skipping"
+        else
+          echo "$resource $name in namespace $namespace has finalizers $finalizers"
+          new_finalizers=$(jq 'del(.[] | select(. == "finalizer.stack.pulumi.com"))' <<< "$finalizers")
```

Comment:
do we need to be that clever? I thought we can just nuke all finalizers from stacks and updates.

Reply:
some had k8s finalizers that I think prevent deleting the namespaces for example if there's a CR still in that namespace, that could get us into some really weird state, that's why I think it's safer to delete just pulumi finalizers and let the rest of them do their thingy